storage/src/nodestore/mod.rs

```diff
@@ -270,6 +278,45 @@ impl<S: ReadableStorage> NodeStore<MutableProposal, S> {
     }
 }
 
+impl<S: ReadableStorage> NodeStore<MutableProposal, S> {
+    /// Creates a new [`NodeStore`] from a child node.
+    ///
+    /// # Errors
+    ///
+    /// Returns a [`FileIoError`] if the child node cannot be read.
+    pub fn from_child(
```

Comment:
Nodestores understanding Child seems non-intuitive. The calculation of the root and the creation of the Nodestore should be separated as well, and this function really shouldn't be doing any I/O.

Maybe this is `from_root(parent: ..., root: Node)` and the caller does the I/O.